### ISSUE TITLE
InputComponent and Reader

### ISSUE THREAD

**Issue by tperroin** [Talend/components#434]
Hello guys !

In theory I have to create an `Input` component and I would like to get my schema value from his `Reader` class. 

Can we pass `IndexRecord` to a class?

Is that a good practice? Have you already tied that?

Thank you

**Comment by tperroin** [comment]
Hello everyone !

I succeed to create a Reader to get the return data from my component !

As @gonchar-ivan told me, I used the widget: `TABLE_WIDGET_TYPE` to work with _properties_ rather than _schema_.

So my flow looks like that for now:

![image](https://cloud.githubusercontent.com/assets/1137581/22111587/136e4e5e-de60-11e6-9f55-9916a7a525f9.png)

With a dynamic form for table properties, corresponding to my previous input schema.

![image](https://cloud.githubusercontent.com/assets/1137581/22111627/41cd6dca-de60-11e6-933f-189ae5adb4d8.png)

I have issues with `messages.properties` but it's ok. 

My reader functions are classic ones.

Now I would like to manage my rejects, in fact my APIs errors.

Does a `InputComponent` can produce a reject output schema?

Regards,

Thibault
